3-With-Intune MAM/IntuneClient/active-directory-xamarin-intune/MainPage.xaml.cs

```diff
@@ -48,28 +48,31 @@ async void btnAcquireToken_Clicked(System.Object sender, System.EventArgs e)
                     // if the scope requires App Protection Policy,  IntuneAppProtectionPolicyRequiredException is thrown.
                     // Perform registration operation here and then does the silent token acquisition
                     var intuneConnector = DependencyService.Get<IIntuneMAMConnector>(DependencyFetchTarget.GlobalInstance);
-                    await intuneConnector.DoMAMRegisterAsync(exProtection).ContinueWith(async (arg) =>
-                    {
-                        // Now the device is registered, perform token acquisition
-                        try
-                        {
-                            // if no MFA Policy is present, the silent should work.
-                            result = await PCAWrapper.Instance.AcquireTokenSilentAsync(Scopes).ConfigureAwait(false);
+                    _ = Task.Factory.StartNew(async () =>
+                                            { 
+                                                await intuneConnector.DoMAMRegisterAsync(exProtection).ContinueWith(async (arg) =>
+                                                {
```

Comment:
nit: seems like a big block of code here, maybe add a method for it?

Reply:
Yes. This is long block of code. But it visually shows how code flows in different conditions.